tests/transport/test_locality.cpp

```diff
@@ -0,0 +1,72 @@
+/* 
+ * GridTools
+ * 
+ * Copyright (c) 2014-2020, ETH Zurich
+ * All rights reserved.
+ * 
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * 
+ */
+#include <ghex/threads/none/primitives.hpp>
+#include <vector>
+#include <iomanip>
+#include <utility>
+#include <unistd.h>
+#include <limits.h>
+#include <cstring>
+
+#include <gtest/gtest.h>
+
+#ifdef GHEX_TEST_USE_UCX
+#include <ghex/transport_layer/ucx/context.hpp>
+using transport = gridtools::ghex::tl::ucx_tag;
+#else
+#include <ghex/transport_layer/mpi/context.hpp>
+using transport = gridtools::ghex::tl::mpi_tag;
+#endif
+
+using threading = gridtools::ghex::threads::none::primitives;
+using context_type = gridtools::ghex::tl::context<transport, threading>;
+
+// test locality by collecting all local ranks
+TEST(locality, enumerate) {
+    auto context_ptr = gridtools::ghex::tl::context_factory<transport,threading>::create(1, MPI_COMM_WORLD);
+    auto& context = *context_ptr;
+    auto comm = context.get_communicator(context.get_token());
+
+    // test self
+    EXPECT_TRUE( comm.is_local(comm.rank()) );
+
+    // check for symmetry
+    std::vector<int> local_ranks(comm.size());
+    std::vector<char> my_host_name(HOST_NAME_MAX+1,0);
+    std::vector<char> other_host_name(HOST_NAME_MAX+1,0);
```

Comment:
```suggestion
    // host names must be contained in a message-compatible data
    std::vector<char> my_host_name(HOST_NAME_MAX+1,0);
    std::vector<char> other_host_name(HOST_NAME_MAX+1,0);
```
Just adding the comment, since I spent some time on why this was a vector and not just a `char[]`

Reply:
no reason, will change